conans/client/tools/system_pm.py

```diff
@@ -126,6 +126,49 @@ def install(self, packages, update=True, force=False, arch_names=None):
             self.update()
         self._install_any(packages)
 
+    def install_packages(self, packages, update=True, force=False, arch_names=None):
+        """ Get the system package tool install command and install all packages or variants.
+            Inputs:
+            "pkg-variant1"  # (1) String input: Install only one package
+            ["pkg-variant1", "otherpkg", "thirdpkg"] # (2) List of strings: all must be installed
+            [("pkg-variant1", "pkg-variant2"), "otherpkg", "thirdpkg"] # (3) List with alternatives
+
+        :param packages: String/List/Tuple with all packages to be installed e.g. "libusb-dev libfoobar-dev"
```

Comment:
If we are accepting the syntax `install_packages("pkg1 pkg2")`, please add it to the docstring.

But, IMHO, we should avoid that syntax because it can be confusing: what does it mean?
 * will it install `pkg1` and `pkg2` or just one of them?
 * what if I use `install_packages(["pkg1 pkg2", "pkg3 pkg4"])`?

Reply:
> If we are accepting the syntax install_packages("pkg1 pkg2"), please add it to the docstring.

Sure, indeed the docstring is outdated since the last commit.

> what if I use install_packages(["pkg1 pkg2", "pkg3 pkg4"])?

I would like to avoid it, but is totally valid for the current behavior, it will install all packages.

Reply:
All packages? pkg1, pkg2, pkg3 and pkg4? It doesn't match the expectations when you use a _list of tuples_...

Reply:
`install_packages(["pkg1 pkg2", "pkg3 pkg4"])` will install all packages pkg1, pkg2, pkg3 and pkg4, yes. Variants should be a nested list or a tuple in a list, like:

`install_packages([["pkg1 pkg2"], "pkg3 pkg4"])`

Reply:
IMHO, it is very confusing if `install_packages(["pkg1 pkg2", "pkg3 pkg4"])` installs all packages, but `install_packages([("pkg1", "pkg2"), ("pkg3", "pkg4")])` install only some of them. I'd totally forbid (raise) the syntax with blank-separated packages (`pkg1 pkg2`).

...but this is my point of view, what do other reviewers think about it? cc/ @memsharded

Reply:
I agree with @jgsogo 

Space separated list introduce another way to do the same thing, but the semantics are now not clear and difficult to dissambiguate. I think it is better to have the following logic:

- Each argument could be either a string, or a list/tuple.
- If a string, that is a package name to install
- If a list/tuple, that is a set of alternatives, install the first successful one.

Reply:
> If a string, that is a package name to install

Considering the previous case, it's a string and contains more than one package. It's totally valid for install(), it will install a variant, but only because install() only installs variants, not all packages. I'll update to set is as invalid, but it won't follow the same design as did for install().

Reply:
Done, now ["foo bar"] and "foo bar" are considered invalid. Actually doesn't matter if you pass a string or list, it's convert to a list at the first step.